### PR TITLE
Refactor Observe example

### PR DESCRIPTION
**Pull request by kolibril13**
I renamed some variables, changed the traitlets variable type and put the check_is_even logic in a separete function. I think this makes this example easier to understand.
What do you think @paddymul?

### CODE REVIEW

**Comment by paddymul** [comment]
I like it.  Maybe name this EvenOddWidget?

Generally I like examples to be small and simple for each facet of functionality.  If I was building with this and I ran into a bug, I would scan for examples to find one that was doing something close to what I wanted, then change that example closer to my usecase, or try to spot what I was doing wrong.

Examples that require a lot of baggage dependencies are a slower to understand.   This is good as is.

As a side note, I know that with traitlets you get the change event, unless I'm specifically interested in the delta, I choose to just access the value on the instance `self.count` vs reaching into the change event `change['new']`.  Not sure about the recommended practice.

**Comment by kolibril13** [comment]
> Generally I like examples to be small and simple for each facet of functionality

I absolutely share this philosophy!

What I really like is to have minimal boilerplate examples that showcase exactly one feature, which I can then use to build  my scripts for my usecases.

Regarding the traitlets: I think `self.count` looks more familiar, and is easier to understand, so I'd go with that.

File: examples/Observe_example.ipynb [hunk 1]
```diff
@@ -2,75 +2,55 @@
...
     "    #it is key that we also take the change argument, otherwise a hard to find python error will be thrown\n",
     "    @observe('count')\n",
     "    def _observe_count(self, change):\n",
-    "        new_val = self.count\n",
-    "        if new_val %2 == 0:\n",
-    "            self.is_even = True\n",
-    "        else:\n",
-    "            self.is_even = False\n",
-    "    \n",
+    "        self.message = check_is_even(self.count)\n",
```
Comment (maartenbreddels) [inline examples/Observe_example.ipynb:37]: I think the `check_is_even` should not return a string, but a boolean, the widget should translate that to a string, sth like:
```suggestion
    "        self.message = "Yes ✅✅✅\" if check_is_even(self.count) else "No 🧊🧊🧊\"\n",
```

**Comment by kolibril13** [comment]
> check_is_even should not return a string, but a boolean
this is adjusted now!

**Comment by maartenbreddels** [comment]
image

oops, code suggestions for notebooks are not ideal ;), sorry for breaking it, I think you need to manually edit the json file

**Comment by paddymul** [comment]
What is the suggested way to do PRs on notebooks.  I have heard of a bunch of projects around this.  What are your thoughts?

**Comment by kolibril13** [comment]
Yes, some invalid notebook syntax was introduced by [14719dc](https://github.com/widgetti/ipyreact/pull/14/commits/14719dc8c635f0a6031504cfc78029e8f295d4f1), but I manually fixed that, now the notebook div should look good again.

image
Tip:
You can enable "Rich Jupyter Notebook Diffs" in Github.
It's still beta, but already works quite well:
image

**Comment by kolibril13** [comment]
@maartenbreddels : This is now ready to be merged.

**Pull request merged**